### ISSUE TITLE
attribute 'model' removed from hparams because it cannot be pickled

### ISSUE THREAD

**Issue by VisionH222** [Lightning-AI/pytorch-lightning#19573]
### Bug description

D:\Anaconda\envs\VH\lib\site-packages\pytorch_lightning\utilities\parsing.py:104: UserWarning: attribute 'model' removed from hparams because it cannot be pickled
 rank_zero_warn(f"attribute '{k}' removed from hparams because it cannot be pickled")
 D:\Anaconda\envs\VH\lib\site-packages\pytorch_lightning\utilities\parsing.py:104: UserWarning: attribute 'loss_fn' removed from hparams because it cannot be pickled
 rank_zero_warn(f"attribute '{k}' removed from hparams because it cannot be pickled")

### What version are you seeing the problem on?

v1.8

### Error messages and logs

```
# Error messages and logs here please
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 2.0):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):
```

</details>

**Comment by awaelchli** [comment]
@VisionH222 I cannot tell exactly what is happening because you didn't post the code. But this is a warning and must mean you have passed an unpicklable argument to your LightningModule, and this simply can't be saved using `save_hyperparameters()`. If you want to avoid the warning, set the ignore parameter:

```py
class MyLightningModule(LightningModule):
    def __init__(self, loss_fn):
        super().__init__()
        self.save_hyperparameters(ignore=["loss_fn"])
```

**Comment by awaelchli** [comment]
Yes, if you want to load from the checkpoint, you will have to [pass the missing argument](https://lightning.ai/docs/pytorch/stable/common/checkpointing_basic.html#initialize-with-other-parameters):

```
LitLungTumorSegModel.load_from_checkpoint(path, loss_fn=...)
```

If you want to load it for inference where you don't need the loss function, maybe make your `loss_fn` argument optional.
